src/test/java/org/folio/rest/impl/GroupFiscalYearSummariesTest.java

```diff
@@ -0,0 +1,111 @@
+package org.folio.rest.impl;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.folio.rest.util.MockServer.addMockEntry;
+import static org.folio.rest.util.TestEntities.BUDGET;
+import static org.folio.rest.util.TestEntities.GROUP_FUND_FISCAL_YEAR;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.core.Is.is;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.folio.rest.jaxrs.model.Budget;
+import org.folio.rest.jaxrs.model.GroupFiscalYearSummary;
+import org.folio.rest.jaxrs.model.GroupFiscalYearSummaryCollection;
+import org.folio.rest.jaxrs.model.GroupFundFiscalYear;
+import org.folio.rest.jaxrs.resource.FinanceGroupFiscalYearSummaries;
+import org.folio.rest.util.HelperUtils;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+
+public class GroupFiscalYearSummariesTest extends ApiTestBase {
+
+  private static final Logger logger = LoggerFactory.getLogger(GroupFiscalYearSummariesTest.class);
+
```

Comment:
As per story overview, a query can have `ledgerId` but I don’t see any tests which handle that. Also, implementation of `GroupFiscalYearSummariesHelper#getGroupFiscalYearSummaries` does not have any logic to handle `ledgerId`. How will this use-case work?

Reply:
Hi @javalkar-varun ,
this should work as ordinary CQL query. Both GroupFundFiscalYear and Budgets have access to Fund's parameters for sorting/filtering through foreign key cross table index queries mechanism. That means query ...?query=fung.ledgerId={{id}} returns us all the budgets and group fund fiscal years associated with funds that contain ledgerId={{id}}. After that logic regroup/recalculates all the retrieved budgets to group fiscal year summaries. As far as queries in the unit tests I think it doesn't make sense to introduce searching by ledgerId in the MockServer. It is enough to validate API with arbitrary parameter just to cover all the logic. And moreover UI is going to use searching by groupId.

Reply:
I think it is fine then. Thanks @khandramai.